src/reducers/AddSiteReducer.js

```diff
@@ -74,14 +95,32 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, facilitiesOption};
 
         case ADD_SITE_FIELDS_RESET:
-            return INITIAL_STATE;
+            return {...INITIAL_STATE, addSiteModalVisible: true};
 
         case CURRENT_LOCATION_UPDATED:
             const {currentLocation: {longitude, latitude}} = payload;
-            const stringLong =  _.toString(longitude);
+            const stringLong = _.toString(longitude);
             const stringLat = _.toString(latitude);
 
-            return {...state, longitudeText: stringLong, latitudeText: stringLat};
+            return {
+                ...state,
+                longitudeText: stringLong,
+                latitudeText: stringLat,
+                readyLatitude: latitude,
+                readyLongitude: longitude
```

Comment:
When we grab location straight from GPS, we can just pass along the lat/long that were returned (since they're already numbers)